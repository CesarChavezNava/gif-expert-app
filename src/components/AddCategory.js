import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    const initialState = '';
    const [inputValue, setInputValue] = useState(initialState);

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length > 2) {
            setCategories( categories => [...categories, inputValue]);
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input type="text" value={inputValue} onChange={handleInputChange} />
            <h2>Add Category</h2>
        </form>
    )
}

AddCategory.prototype = {
    setCategories: PropTypes.func.isRequired
}
