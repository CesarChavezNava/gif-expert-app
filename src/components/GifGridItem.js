import React from 'react'

export const GifGridItem = ( { title, url} ) => {
    return (
        <div className="column is-4">
            <div className="card">
                <div className="card-image">
                    <figure className="image is-4by3">
                        <img src={url} alt={title} />
                    </figure>
                </div>
                <div className="card-content">
                    <div className="content">
                        <p>{title}</p>
                    </div>
                </div>
            </div>
        </div>
    )
}
