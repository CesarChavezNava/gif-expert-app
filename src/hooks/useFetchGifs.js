import { useState, useEffect } from "react"
import { getGifts } from "../helpers/getGifts";

export const useFetchGifs = (category) => {
    const initialState = { data: [], loading: true };
    const [state, setState] = useState(initialState)

    useEffect(() => {
        getGifts(category).then( images => {
            setState({
                data: images,
                loading: false
            });
        });
    }, [category])

    return state;
}
