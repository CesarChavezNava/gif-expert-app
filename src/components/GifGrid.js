import React from 'react'
import PropTypes from 'prop-types';

import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <div className="container is-fluid">
            { loading && <progress className="progress is-small is-primary" max="100">15%</progress> }
           <h3 className="subtitle">{category}</h3> 
           <div className="columns">
               {
                   images.map((image) => (
                    <GifGridItem key={image.id} {...image} />
                   ))
               }
           </div>
        </div>
    )
}

GifGrid.prototype = {
    category: PropTypes.string.isRequired
}