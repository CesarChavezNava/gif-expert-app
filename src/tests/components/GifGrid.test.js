import React from 'react';
import { shallow } from 'enzyme';

import '@testing-library/jest-dom';

import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas de GifGrid', () => {
    const category = 'Overwatch';
    
    test('Debe mostrar el componente correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan imÃ¡genes', () => {
        const gifts = [{
            id: 'qaswdefrgthyjuk',
            url: 'https://localhost/imagen.jpg',
            title: 'culaquier cosa'
        },
        {
            id: '1223435436567567',
            url: 'https://localhost/imagen2.jpg',
            title: 'culaquier cosa 2'
        }]

        useFetchGifs.mockReturnValue({
            data: gifts,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('progress').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifts.length);
    });
});
