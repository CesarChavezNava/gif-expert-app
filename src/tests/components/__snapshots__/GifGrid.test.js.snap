// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGrid Debe de mostrar items cuando se cargan im√°genes 1`] = `
<div
  className="container is-fluid"
>
  <h3
    className="subtitle"
  >
    Overwatch
  </h3>
  <div
    className="columns"
  >
    <GifGridItem
      id="qaswdefrgthyjuk"
      key="qaswdefrgthyjuk"
      title="culaquier cosa"
      url="https://localhost/imagen.jpg"
    />
    <GifGridItem
      id="1223435436567567"
      key="1223435436567567"
      title="culaquier cosa 2"
      url="https://localhost/imagen2.jpg"
    />
  </div>
</div>
`;

exports[`Pruebas de GifGrid Debe mostrar el componente correctamente 1`] = `
<div
  className="container is-fluid"
>
  <progress
    className="progress is-small is-primary"
    max="100"
  >
    15%
  </progress>
  <h3
    className="subtitle"
  >
    Overwatch
  </h3>
  <div
    className="columns"
  />
</div>
`;
